File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.py
a = input().strip()
boom = input().strip()
ary = []
for i in range(len(a)):
    ary.append(a[i])
    if len(a) >= len(boom):
        tmp = "".join(ary[-len(boom):])
        if tmp == boom:
            cnt = 0
            while cnt < len(boom):
                ary.pop()
                cnt += 1
if (len(ary) == 0):
    print("FRULA")
else:
    print("".join(ary))

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.cc
#include <iostream>
#include <string>
using namespace std;

string ans = "", x, boom;
int cnt;
int main(void)
{
	cin >> x >> boom;

	for (int i = 0; i < x.size(); i++)
	{
		ans += x[i];
		if (ans.size() >= boom.size())
		{
			if (ans.substr(ans.size() - boom.size(), boom.size()) == boom)
			{
				cnt = 0;
				while (cnt != boom.size())
				{
					ans.pop_back();
					cnt++;
				}
			}
		}
	}

	if (!ans.size())
		cout << "FRULA";
	else
		cout << ans;

	return 0;
}